Main.py

```diff
@@ -1,62 +1,249 @@
+        #this if statement handles returning the rate as float
+        if queryBool is True:
+            print (f'Currency is {currency}')
+            return float(APIService.currencyRate(currency,cur))
+
+        #if the currency doesn't exist it gives the user an error
+        else:
+            print ('Currency does not exist.')
+
+
+"""
+Mock Data Handler Class in case API not working
+"""
+class MockDataService(AbstractCurrencyService):
+    #will need to overwrite csv every time
+    def get_currency_data():
+        print("API not connected-out of date exchange rates in use")
+       
+    def getCurrency():
+        print("Please type in the abbreviation of the currency in which the donation will be received")
+        donorCurrency= input()
+        donCurrCaps = donorCurrency.upper()
+        return donCurrCaps
+    
+    def currencyRate(dc, cur):
+    #sql query to get rate inserting currency into where clause
+        res=cur.execute("SELECT rate FROM rates WHERE currency = ?", (dc,))
+        x = res.fetchall()
+
+        #x is a list of tuples, need to separate out rate as float
+        conversionRate = x[0][0]
+        return conversionRate
+    
+    def check_for_currency(dc, cur)->bool:
+        #sql query to return a boolean
+        res=cur.execute("SELECT rate FROM rates WHERE currency = ?", (dc,))
+        queryBool = res.fetchone() is not None
+
+        #this if statement handles returning the rate as float
+        if queryBool is True:
+            print (f'Currency is {dc}')
+            return float(APIService.currencyRate(dc,cur))
+
+        #if the currency doesn't exist it gives the user an error
+        else:
+            print ('Currency does not exist.')
+
+
+    
+"""
+Main
+"""
+#1)Check if converting currency
+# 2)check if update datetime align
+#  A)if yes, call api and get response
+#    1)if api response is 200, update the db and csv and use API service
+#    2)if api response is NOT 200 use mock data service (which will just provide un-updated db with warning)
+#  B)if no, do not call api and proceed with program
+
 
 #gofundme fees, 2.9% + $.30usd
 feePercentage=.029
 flatFeeAddOn=.3 #will need to convert this when converting currency
-#amountIntended
-#floatIntended
+
 print("Converting between any currencies? Y or N")
 converting=input()
 #if yes
 if(converting == "Y" or converting =="y"):
-    #so far im only doing USD as the donator currency
-        #print("Please type in the abbreviation of the currency in which you use")
-        #print("eg. USD, GBP, IQD, etc")
-        #userCurrency=input()
-    print("Please type in the abbreviation of the currency in which the donation will be received")
-    donationCurrency=input()
+    con = sqlite3.connect("currencyRates.db")
+    cur = con.cursor()
+    tf=APIService.compareDateTime()
+    if(tf==True):
+        #call aip
+        apiValidation=APIService.get_currency_data(cur, con) #db is now up to date
+        if(apiValidation.status_code==200):
```

Comment:
Instead of having this real api/mock switch as part of error handling, I think it would make sense to keep a separate variable that enforces whether to use the mock/api handlers.

Reason for this is that if the intention is to use the real API, we don't want failures to go unnoticed and the user getting 'mock' data when that was not the intention.

Definitely the error handling here makes sense, but I would just exit the code by printing an error statement, or by using the `throw` statement to raise and exception.